Clang Warnings

So many clang warnings lately:
- OpenSSL 3 deprecation warnings
- thread-safety-analysis

<Details>

```
[40/256] Building CXX object CMakeFiles/engine-shared.dir/src/engine/shared/jobs.cpp.o
src/engine/shared/jobs.cpp:113:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/shared/jobs.cpp:110:2: note: mutex acquired here
        }
        ^
src/engine/shared/jobs.cpp:113:1: warning: mutex 'm_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/shared/jobs.cpp:110:2: note: mutex acquired here
        }
        ^
2 warnings generated.
[64/256] Building CXX object CMakeFiles/engine-shared.dir/src/base/hash_openssl.cpp.o
src/base/hash_openssl.cpp:6:2: warning: 'SHA256_Init' is deprecated [-Wdeprecated-declarations]
        SHA256_Init(ctxt);
        ^
/usr/include/openssl/sha.h:73:1: note: 'SHA256_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Init(SHA256_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
                                                ^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
                                                   ^
src/base/hash_openssl.cpp:11:2: warning: 'SHA256_Update' is deprecated [-Wdeprecated-declarations]
        SHA256_Update(ctxt, data, data_len);
        ^
/usr/include/openssl/sha.h:74:1: note: 'SHA256_Update' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Update(SHA256_CTX *c,
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
                                                ^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
                                                   ^
src/base/hash_openssl.cpp:17:2: warning: 'SHA256_Final' is deprecated [-Wdeprecated-declarations]
        SHA256_Final(result.data, ctxt);
        ^
/usr/include/openssl/sha.h:76:1: note: 'SHA256_Final' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Final(unsigned char *md, SHA256_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
                                                ^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
                                                   ^
src/base/hash_openssl.cpp:23:2: warning: 'MD5_Init' is deprecated [-Wdeprecated-declarations]
        MD5_Init(ctxt);
        ^
/usr/include/openssl/md5.h:49:1: note: 'MD5_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int MD5_Init(MD5_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
                                                ^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
                                                   ^
src/base/hash_openssl.cpp:28:2: warning: 'MD5_Update' is deprecated [-Wdeprecated-declarations]
        MD5_Update(ctxt, data, data_len);
        ^
/usr/include/openssl/md5.h:50:1: note: 'MD5_Update' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int MD5_Update(MD5_CTX *c, const void *data, size_t len);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
                                                ^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
                                                   ^
src/base/hash_openssl.cpp:34:2: warning: 'MD5_Final' is deprecated [-Wdeprecated-declarations]
        MD5_Final(result.data, ctxt);
        ^
/usr/include/openssl/md5.h:51:1: note: 'MD5_Final' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int MD5_Final(unsigned char *md, MD5_CTX *c);
^
/usr/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
#   define OSSL_DEPRECATEDIN_3_0                OSSL_DEPRECATED(3.0)
                                                ^
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
                                                   ^
6 warnings generated.
[91/256] Building CXX object CMakeFiles/engine-shared.dir/src/base/system.cpp.o
src/base/system.cpp:665:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/base/system.cpp:664:14: note: mutex acquired here
        return aio->error;
                    ^
src/base/system.cpp:665:1: warning: mutex 'aio->lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/base/system.cpp:664:14: note: mutex acquired here
        return aio->error;
                    ^
src/base/system.cpp:685:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/base/system.cpp:683:2: note: mutex acquired here
        }
        ^
src/base/system.cpp:685:1: warning: mutex 'aio->lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/base/system.cpp:683:2: note: mutex acquired here
        }
        ^
src/base/system.cpp:701:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/base/system.cpp:698:2: note: mutex acquired here
        }
        ^
src/base/system.cpp:701:1: warning: mutex 'aio->lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/base/system.cpp:698:2: note: mutex acquired here
        }
        ^
6 warnings generated.
[135/256] Building CXX object CMakeFiles/DDNet.dir/src/engine/client/serverbrowser_http.cpp.o
src/engine/client/serverbrowser_http.cpp:147:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/serverbrowser_http.cpp:147:1: note: mutex acquired here
src/engine/client/serverbrowser_http.cpp:147:1: warning: mutex 'm_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/serverbrowser_http.cpp:147:1: note: mutex acquired here
src/engine/client/serverbrowser_http.cpp:171:17: warning: expecting mutex 'ls.m_Lock' to be held at start of each loop [-Wthread-safety-analysis]
        for(int i = 0; i < m_pData->m_NumUrls; i++)
                       ^
src/engine/client/serverbrowser_http.cpp:181:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:171:17: warning: expecting mutex 'ls.m_Lock' to be held at start of each loop [-Wthread-safety-analysis]
        for(int i = 0; i < m_pData->m_NumUrls; i++)
                       ^
src/engine/client/serverbrowser_http.cpp:199:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:171:17: warning: expecting mutex 'm_Lock' to be held at start of each loop [-Wthread-safety-analysis]
        for(int i = 0; i < m_pData->m_NumUrls; i++)
                       ^
src/engine/client/serverbrowser_http.cpp:199:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:171:41: warning: expecting mutex 'ls.m_Lock' to be held at start of each loop [-Wthread-safety-analysis]
        for(int i = 0; i < m_pData->m_NumUrls; i++)
                                               ^
src/engine/client/serverbrowser_http.cpp:199:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:197:15: warning: cannot call function 'CLockScope' while mutex 'm_Lock' is held [-Wthread-safety-analysis]
                        CLockScope ls(m_Lock);
                                   ^
src/engine/client/serverbrowser_http.cpp:197:15: warning: acquiring mutex 'm_Lock' requires negative capability '!m_Lock' [-Wthread-safety-negative]
src/engine/client/serverbrowser_http.cpp:197:15: warning: acquiring mutex 'm_Lock' that is already held [-Wthread-safety-analysis]
src/engine/client/serverbrowser_http.cpp:181:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:247:1: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/client/serverbrowser_http.cpp:181:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:247:1: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/client/serverbrowser_http.cpp:199:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:247:1: warning: mutex 'm_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/client/serverbrowser_http.cpp:199:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:247:1: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/client/serverbrowser_http.cpp:181:3: note: mutex acquired here
                }
                ^
src/engine/client/serverbrowser_http.cpp:247:1: warning: mutex 'm_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/client/serverbrowser_http.cpp:181:3: note: mutex acquired here
                }
                ^
14 warnings generated.
[140/256] Building CXX object CMakeFiles/DDNet.dir/src/engine/client/updater.cpp.o
src/engine/client/updater.cpp:59:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:59:1: note: mutex acquired here
src/engine/client/updater.cpp:59:1: warning: mutex 'm_pUpdater->m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:59:1: note: mutex acquired here
src/engine/client/updater.cpp:117:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:117:1: note: mutex acquired here
src/engine/client/updater.cpp:117:1: warning: mutex 'm_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:117:1: note: mutex acquired here
src/engine/client/updater.cpp:123:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:122:9: note: mutex acquired here
        return m_State;
               ^
src/engine/client/updater.cpp:123:1: warning: mutex 'm_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:122:9: note: mutex acquired here
        return m_State;
               ^
src/engine/client/updater.cpp:129:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:129:1: note: mutex acquired here
src/engine/client/updater.cpp:129:1: warning: mutex 'm_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:129:1: note: mutex acquired here
src/engine/client/updater.cpp:135:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:134:9: note: mutex acquired here
        return m_Percent;
               ^
src/engine/client/updater.cpp:135:1: warning: mutex 'm_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/client/updater.cpp:134:9: note: mutex acquired here
        return m_Percent;
               ^
10 warnings generated.
[150/256] Building CXX object CMakeFiles/DDNet.dir/src/engine/client/video.cpp.o
src/engine/client/video.cpp:452:9: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
        while(!pThreadData->m_Finished)
               ^
src/engine/client/video.cpp:476:5: note: mutex acquired here
                                }
                                ^
src/engine/client/video.cpp:452:9: warning: mutex 'g_WriteLock' is not held on every path through here [-Wthread-safety-analysis]
        while(!pThreadData->m_Finished)
               ^
src/engine/client/video.cpp:476:5: note: mutex acquired here
                                }
                                ^
src/engine/client/video.cpp:534:9: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
        while(!pThreadData->m_Finished)
               ^
src/engine/client/video.cpp:557:5: note: mutex acquired here
                                }
                                ^
src/engine/client/video.cpp:534:9: warning: mutex 'g_WriteLock' is not held on every path through here [-Wthread-safety-analysis]
        while(!pThreadData->m_Finished)
               ^
src/engine/client/video.cpp:557:5: note: mutex acquired here
                                }
                                ^
4 warnings generated.
[222/256] Building CXX object CMakeFiles/DDNet-Server.dir/src/engine/server/register.cpp.o
src/engine/server/register.cpp:322:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:277:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:322:1: warning: mutex 'm_pShared->m_pGlobal->m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:277:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:322:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:316:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:322:1: warning: mutex 'm_pShared->m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:316:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:388:1: warning: mutex 'ls.m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:388:1: note: mutex acquired here
src/engine/server/register.cpp:388:1: warning: mutex 'm_pShared->m_Lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:388:1: note: mutex acquired here
src/engine/server/register.cpp:478:1: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:477:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:478:1: warning: mutex 'm_pShared->m_pGlobal->m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:477:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:478:1: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:468:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:478:1: warning: mutex 'm_pShared->m_pGlobal->m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:468:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:478:1: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:460:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:478:1: warning: mutex 'm_pShared->m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:460:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:715:1: warning: mutex 'ls.m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:671:2: note: mutex acquired here
        }
        ^
src/engine/server/register.cpp:715:1: warning: mutex 'm_pGlobal->m_Lock' is not held on every path through here [-Wthread-safety-analysis]
}
^
src/engine/server/register.cpp:671:2: note: mutex acquired here
        }
        ^
14 warnings generated.

```

</Details>

--- a/src/base/lock_scope.h
+++ b/src/base/lock_scope.h
@@ -6,7 +6,7 @@
 class SCOPED_CAPABILITY CLockScope
 {
 public:
-	CLockScope(LOCK Lock) ACQUIRE(Lock, m_Lock) REQUIRES(!Lock, !m_Lock) :
+	CLockScope(LOCK Lock) ACQUIRE(Lock, m_Lock) :
 		m_Lock(Lock)
 	{
 		lock_wait(m_Lock);

--- a/src/tools/map_create_pixelart.cpp
+++ b/src/tools/map_create_pixelart.cpp
@@ -6,25 +6,25 @@
 #include <engine/storage.h>
 #include <game/mapitems.h>
 
-bool CreatePixelArt(const char[][64], const int[], const int[], int[], const bool[]);
+bool CreatePixelArt(const char[3][64], const int[2], const int[2], int[2], const bool[2]);
 void InsertCurrentQuads(CDataFileReader &, CMapItemLayerQuads *, CQuad *);
-int InsertPixelArtQuads(CQuad *, int &, const CImageInfo &, const int[], const int[], const bool[]);
+int InsertPixelArtQuads(CQuad *, int &, const CImageInfo &, const int[2], const int[2], const bool[2]);
 
 bool LoadPNG(CImageInfo *, const char *);
-bool OpenMaps(const char[][64], CDataFileReader &, CDataFileWriter &);
+bool OpenMaps(const char[2][64], CDataFileReader &, CDataFileWriter &);
 void SaveOutputMap(CDataFileReader &, CDataFileWriter &, CMapItemLayerQuads *, int, CQuad *, int);
 
-CMapItemLayerQuads *GetQuadLayer(CDataFileReader &, const int[], int *);
-CQuad CreateNewQuad(float, float, int, int, const uint8_t[], const int[]);
+CMapItemLayerQuads *GetQuadLayer(CDataFileReader &, const int[2], int *);
+CQuad CreateNewQuad(float, float, int, int, const uint8_t[4], const int[2]);
 
-bool GetPixelClamped(const CImageInfo &, int, int, uint8_t[]);
-bool ComparePixel(const uint8_t[], const uint8_t[]);
-bool IsPixelOptimizable(const CImageInfo &, int, int, const uint8_t[], const bool[]);
+bool GetPixelClamped(const CImageInfo &, int, int, uint8_t[4]);
+bool ComparePixel(const uint8_t[4], const uint8_t[4]);
+bool IsPixelOptimizable(const CImageInfo &, int, int, const uint8_t[4], const bool[]);
 void SetVisitedPixels(const CImageInfo &, int, int, int, int, bool[]);
 
 int GetImagePixelSize(const CImageInfo &);
-int FindSuperPixelSize(const CImageInfo &, const uint8_t[], int, int, int, bool[]);
-void GetOptimizedQuadSize(const CImageInfo &, int, const uint8_t[], int, int, int &, int &, bool[]);
+int FindSuperPixelSize(const CImageInfo &, const uint8_t[4], int, int, int, bool[]);
+void GetOptimizedQuadSize(const CImageInfo &, int, const uint8_t[4], int, int, int &, int &, bool[]);
 
 int main(int argc, const char **argv)
 {

--- a/src/tools/map_replace_area.cpp
+++ b/src/tools/map_replace_area.cpp
@@ -23,36 +23,36 @@ struct MapObject // quad pivot or tile layer
 	float m_aaExtendedArea[2][2]; // extended with parallax
 };
 
-bool ReplaceArea(IStorage *, const char[][64], const float[][2][2]);
-bool OpenMaps(IStorage *, const char[][64], CDataFileReader[], CDataFileWriter &);
+bool ReplaceArea(IStorage *, const char[3][64], const float[][2][2]);
+bool OpenMaps(IStorage *, const char[3][64], CDataFileReader[2], CDataFileWriter &);
 void SaveOutputMap(CDataFileReader &, CDataFileWriter &);
-bool CompareLayers(const char[][64], CDataFileReader[]);
-void CompareGroups(const char[][64], CDataFileReader[]);
+bool CompareLayers(const char[3][64], CDataFileReader[2]);
+void CompareGroups(const char[3][64], CDataFileReader[2]);
 const CMapItemGroup *GetLayerGroup(CDataFileReader &, int);
 
-void ReplaceAreaTiles(CDataFileReader[], const float[][2][2], const CMapItemGroup *[], CMapItemLayer *[]);
-void RemoveDestinationTiles(CMapItemLayerTilemap *, CTile *, float[][2]);
-void ReplaceDestinationTiles(CMapItemLayerTilemap *[], CTile *[], float[][2][2]);
-bool AdaptVisibleAreas(const float[][2][2], const MapObject[], float[][2][2]);
-bool AdaptReplaceableAreas(const float[][2][2], const float[][2][2], const MapObject[], float[][2][2]);
+void ReplaceAreaTiles(CDataFileReader[2], const float[][2][2], const CMapItemGroup *[2], CMapItemLayer *[2]);
+void RemoveDestinationTiles(CMapItemLayerTilemap *, CTile *, float[2][2]);
+void ReplaceDestinationTiles(CMapItemLayerTilemap *[2], CTile *[2], float[2][2][2]);
+bool AdaptVisibleAreas(const float[2][2][2], const MapObject[2], float[2][2][2]);
+bool AdaptReplaceableAreas(const float[2][2][2], const float[2][2][2], const MapObject[2], float[2][2][2]);
 
-void ReplaceAreaQuads(CDataFileReader[], const float[][2][2], const CMapItemGroup *[], CMapItemLayer *[], int);
-bool RemoveDestinationQuads(const float[][2], const CQuad *, int, const CMapItemGroup *, CQuad *, int &);
-bool InsertDestinationQuads(const float[][2][2], const CQuad *, int, const CMapItemGroup *[], CQuad *, int &);
-bool AdaptVisiblePoint(const float[][2][2], const float[][2], const MapObject[], float[]);
+void ReplaceAreaQuads(CDataFileReader[2], const float[][2][2], const CMapItemGroup *[2], CMapItemLayer *[2], int);
+bool RemoveDestinationQuads(const float[2][2], const CQuad *, int, const CMapItemGroup *, CQuad *, int &);
+bool InsertDestinationQuads(const float[2][2][2], const CQuad *, int, const CMapItemGroup *[2], CQuad *, int &);
+bool AdaptVisiblePoint(const float[2][2][2], const float[2][2], const MapObject[2], float[2]);
 
 MapObject CreateMapObject(const CMapItemGroup *, int, int, int, int);
 void SetExtendedArea(MapObject &);
-bool GetVisibleArea(const float[][2], MapObject, float[][2] = 0x0);
-bool GetReplaceableArea(const float[][2], MapObject, float[][2]);
+bool GetVisibleArea(const float[2][2], MapObject, float[2][2] = 0x0);
+bool GetReplaceableArea(const float[2][2], MapObject, float[2][2]);
 
-void GetGameAreaDistance(const float[][2][2], const MapObject[], const float[][2][2], float[]);
-void GetGameAreaDistance(const float[][2][2], const MapObject[], const float[][2], float[]);
-void GetSignificantScreenPos(MapObject, const float[][2], const float[][2], float[]);
-void ConvertToTiles(const float[][2], int[][2]);
+void GetGameAreaDistance(const float[2][2][2], const MapObject[2], const float[2][2][2], float[2]);
+void GetGameAreaDistance(const float[2][2][2], const MapObject[2], const float[2][2], float[2]);
+void GetSignificantScreenPos(MapObject, const float[2][2], const float[2][2], float[2]);
+void ConvertToTiles(const float[2][2], int[2][2]);
 
-bool GetLineIntersection(const float[], const float[], float[] = 0x0);
-bool GetLineIntersection(const float[], float);
+bool GetLineIntersection(const float[2], const float[2], float[2] = 0x0);
+bool GetLineIntersection(const float[2], float);
 void SetInexistent(float *, int);
 bool IsInexistent(const float *, int);
 bool IsInexistent(float);

